// node-wsl/dist/status/statusFactory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createStatus = void 0;
/**
 * Creates the full status object of the wsl installation
 *
 * @param distributions - list of distribution status parsed from `wsl.exe --list --verbose`
 * @returns the wsl status
 */
exports.createStatus = function (distributions) {
    if (!Array.isArray(distributions)) {
        return null;
    }
    var runningDistributions = distributions.filter(function (dist) { return dist.running; });
    var stoppedDistributions = distributions.filter(function (dist) { return dist.stopped; });
    var defaultDistribution = distributions.find(function (dist) { return dist.default; });
    var createdAt = new Date();
    return {
        distributions: distributions,
        runningDistributions: runningDistributions,
        stoppedDistributions: stoppedDistributions,
        defaultDistribution: defaultDistribution,
        total: distributions.length,
        running: runningDistributions.length,
        stopped: stoppedDistributions.length,
        createdAt: createdAt.toISOString(),
    };
};
